HealthGPT/HealthGPT/HealthGPTView.swift

```diff
@@ -4,27 +4,64 @@
 // SPDX-FileCopyrightText: 2023 Stanford University & Project Contributors
 //
 
+import CardinalKitFHIR
+import CardinalKitSecureStorage
 import HealthKit
 import OpenAI
 import SwiftUI
 
 
 struct HealthGPTView: View {
-    @State private var userMessage: String = ""
+    @AppStorage(StorageKeys.openAIModel) var openAIModel = Model.gpt3_5Turbo
+    @EnvironmentObject var secureStorage: SecureStorage<FHIR>
     @State private var messages: [Message] = []
 
+    @State private var showAlert = false
+    @State private var alertText = ""
+
+    @StateObject private var messageManager = MessageManager()
+
     var body: some View {
         NavigationView {
             VStack {
-                ChatView(messages: $messages)
+                ChatView()
+                    .environmentObject(messageManager)
                     .gesture(
                         TapGesture().onEnded {
                             UIApplication.shared.hideKeyboard()
                         }
                     )
-                MessageInputView(userMessage: $userMessage, messages: $messages)
+                MessageInputView()
+                    .environmentObject(messageManager)
             }
             .navigationBarTitle("HealthGPT")
+            .alert(isPresented: $showAlert) {
+                Alert(
+                    title: Text("Alert"),
+                    message: Text(alertText),
+                    dismissButton: .default(Text("OK"))
+                )
+            }
+            .onAppear {
+                // Look for an stored API key
+                var apiKey = ""
+                if let storedApiKey = try? secureStorage.retrieveCredentials(
+                    "openai-api-key",
+                    server: "openai.com"
+                ) {
+                    apiKey = storedApiKey.password
+                } else {
+                    alertText = "Could not find a valid API key."
+                    self.showAlert.toggle()
+                    return
+                }
+
+                // Assign the api key to the message handler
+                messageManager.updateAPIToken(apiKey)
+
+                // Set the OpenAI model in the message handler
+                messageManager.updateOpenAIModel(openAIModel)
```

Comment:
From a code styling perspective and to improve the readability of views, I would suggest to transform this in a private function of the `HealthGPTView` view that is then passed to the `onAppear` view modifier.

Reply:
I agree, this is much cleaner. I've made the change here: [c0b236c](https://github.com/StanfordBDHG/HealthGPT/pull/23/commits/c0b236cb7c25d0a05da857d97cf43d523c2d61cb). Thank you for the suggestion!